piptools/scripts/options.py

```diff
@@ -364,3 +370,28 @@ def _get_default_option(option_name: str) -> Any:
     is_flag=True,
     help="Restrict attention to user directory",
 )
+
+build_deps_for = click.option(
+    "--build-deps-for",
```

Comment:
```suggestion
    "--build-deps-dist",
```

or

```suggestion
    "--build-deps-distribution",
```

This may look more descriptive.

Reply:
We have previously moved away from using the term _distribution_ on the CLI because it may be confused with _distribution package_. But if @webknjaz is happy with the proposed name then I will change it.

Reply:
Perhaps... BTW, is there any discussion about the use of `--build-deps-for`? I'm trying to understand its use case and honestly can't find a reason not to use all distributions.

Reply:
Yeah, if we were to use this term, it'd be something like
```suggestion
    "--dist-build-deps-for",
```
probably.

Reply:
@atugushev some of the discussion is here: https://github.com/jazzband/pip-tools/pull/1681#discussion_r1084778877 there was another one, I think. I requested separate deps for sdist/wheel/editable hooks, which are the only values this setting accepts. I believe that for libraries, it would be useful to have different build environments not influence each other.

@apljungquist I see that the term `distributions` is still being used in the variable and identifier names — maybe we could use `ALL_BUILD_ENV_VARIANTS` and `build_deps_environment`?

Reply:
Yes. I prefer using existing terminology (the `build` package calls it distribution) rather than make up new. What’s the rationale behind calling it _build environment variant_ and _build dependency environment_?

Reply:
@apljungquist

> Yes. I prefer using existing terminology (the `build` package calls it distribution) rather than make up new.

Yeah.. and it's a bit weird. I think that when the were starting the project, there were sdist and wheel which are technically archive formats for bundling distribution packages. But later on, PEP 660 introduced brought in hooks for editable installs which can't correspond to anything resembling a packaging format anymore. https://github.com/pypa/build/blob/699d7a0/src/build/__init__.py#L231-L232 isn't even updated to mention `editable`, for example.
I don't know why it was chosen, maybe they were struggling with naming variables but calling them that is rather confusing. PyPUG describes distribution as follows: 
> > A versioned archive file that contains Python [packages](https://packaging.python.org/en/latest/glossary/#term-Import-Package), [modules](https://packaging.python.org/en/latest/glossary/#term-Module), and other resource files that are used to distribute a [Release](https://packaging.python.org/en/latest/glossary/#term-Release).

_(https://packaging.python.org/en/latest/glossary/#term-Distribution-Package)_

> What’s the rationale behind calling it _build environment variant_ and _build dependency environment_?

I was just trying to come up with something that would reduce confusion. With PEP 517, the front-end creates temporary throw-away build virtualenvs for the build target (sdist, wheel, editable). And those are essentially build environment variants that we're trying to make reproducible here — hence my earlier suggestions.

Does it make more sense in such context?

Reply:
Yeah, you're right. We should probably keep the CLI flag name as is.

Reply:
Consider this resolved 👍🏻